tests/Queue/RedisQueueIntegrationTest.php

```diff
@@ -21,15 +26,36 @@ class RedisQueueIntegrationTest extends PHPUnit_Framework_TestCase
     public function setUp()
     {
         parent::setUp();
+
+        $host = getenv('REDIS_HOST') ?: '127.0.0.1';
+        $port = getenv('REDIS_PORT') ?: 6379;
+
+        if (static::$connectionFailedOnceWithDefaultsSkip) {
+            $this->markTestSkipped('Trying default host/port failed, please set environment variable REDIS_HOST & REDIS_PORT to enable '.__CLASS__);
+
+            return;
+        }
+
         $this->redis = new Database([
             'cluster' => false,
             'default' => [
-                'host' => '127.0.0.1',
-                'port' => 6379,
+                'host' => $host,
+                'port' => $port,
                 'database' => 5,
+                'timeout' => 0.5,
```

Comment:
I'm fine with changing it. The reasoning is that the majority use case for running redis tests is that you have redis either on the same box or in an instance right next to the test runner (like docker), and since it is a test environment, the load is probably very low.  Perhaps this is not the case with some CI environments though-

What is a good value for a test-time timeout?